src/libs/antares/writer/private/in_memory_writer.h

```diff
@@ -0,0 +1,55 @@
+/*
+** Copyright 2007-2024, RTE (https://www.rte-france.com)
+** See AUTHORS.txt
+** SPDX-License-Identifier: MPL-2.0
+** This file is part of Antares-Simulator,
+** Adequacy and Performance assessment for interconnected energy networks.
+**
+** Antares_Simulator is free software: you can redistribute it and/or modify
+** it under the terms of the Mozilla Public Licence 2.0 as published by
+** the Mozilla Foundation, either version 2 of the License, or
+** (at your option) any later version.
+**
+** Antares_Simulator is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+** Mozilla Public Licence 2.0 for more details.
+**
+** You should have received a copy of the Mozilla Public Licence 2.0
+** along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+*/
+#pragma once
+
+#include <mutex>
+#include <string>
+#include <map>
+
+#include <yuni/core/string.h>
+
+#include "antares/writer/i_writer.h"
+#include <antares/benchmarking/DurationCollector.h>
+
+namespace Antares::Solver
+{
+class InMemoryWriter : public IResultWriter
+{
+public:
+    using MapType = std::map<std::string, std::string>;
+    InMemoryWriter(Benchmarking::IDurationCollector& duration_collector);
+    virtual ~InMemoryWriter();
+    void addEntryFromBuffer(const std::string& entryPath, Yuni::Clob& entryContent) override;
+    void addEntryFromBuffer(const std::string& entryPath, std::string& entryContent) override;
+    void addEntryFromFile(const std::string& entryPath, const std::string& filePath) override;
+    void flush() override;
+    bool needsTheJobQueue() const override;
+    void finalize(bool verbose) override;
+
+private:
+    std::mutex pMapMutex;
+    // minizip-ng requires a void* as a zip handle.
+    MapType pEntries;
```

Comment:
Don't we need a way to retrieve the content ?
- either through getters
- or letting the user provide the object where content is written

Reply:
Yes, I agree. I'll provide a getter for the map in this class, so the caller can write this
```cpp
void f(IResultWriter& writer);
InMemoryWriter writer(...);
f(writer);
auto&  map = writer->getMap();
```

What do you think ?